### PR TITLE
Creating new start data collection popup window

### PR DESCRIPTION
**Pull request by krritik**
Created a new start data collection popup window for selecting the path.

The actual data collection will be added in further PR.

Here is the screenshot for same -
image

Mocks for same - https://www.figma.com/file/EC2tJwlwy4RveRYhhGiHP4/Proposal-for-UI-improvement---SQL-Migration-ADS?type=whiteboard&node-id=0-1&t=CB5YVEg0XX4fijyE-0

### CODE REVIEW

File: extensions/sql-migration/src/constants/strings.ts [hunk 1]
```diff
@@ -229,6 +229,10 @@ export const AZURE_RECOMMENDATION_TOOLTIP_IN_PROGRESS = localize('sql.migration.
 export const AZURE_RECOMMENDATION_START = localize('sql.migration.sku.azureRecommendation.start', "Start");
 export const AZURE_RECOMMENDATION_START_POPUP = localize('sql.migration.sku.azureRecommendation.start.popup', "Starting performance data collection...");
 export const AZURE_RECOMMENDATION_STOP_POPUP = localize('sql.migration.sku.azureRecommendation.stop.popup', "Stopping performance data collection...");
+export const AZURE_RECOMMENDATION_DATA_COLLECTION_POPUP_MESSAGE_LABEL = localize('sql.migration.sku.azureRecommendation.data.collection.popup.message.label', "Where do you want to save collected data?");
```
Comment (Ramudaykumar) [inline extensions/sql-migration/src/constants/strings.ts:232]: Generally, we show texts like, choose/select a folder/directory to collect data. Please check with PM whether they want same text.

File: extensions/sql-migration/src/wizard/skuRecommendation/skuDataCollectionToolbar.ts [hunk 2]
```diff
@@ -14,19 +14,40 @@ import * as vscode from 'vscode';
 import { IconPathHelper } from '../../constants/iconPathHelper';
 import * as styles from '../../constants/styles';
 import * as constants from '../../constants/strings';
+import { MigrationStateModel, PerformanceDataSourceOptions } from '../../models/stateMachine';
+
+// TODO - "Change this to actual default path once it is available"
+const Default_PATH_FOR_START_DATA_COLLECTION = "C:\DataPointsCollectionFolder";
```
Comment (Ramudaykumar) [inline extensions/sql-migration/src/wizard/skuRecommendation/skuDataCollectionToolbar.ts:20]: Constant name should be all upper cap.

File: extensions/sql-migration/src/constants/strings.ts (hunk 1 above)
Comment (krritik) [reply extensions/sql-migration/src/constants/strings.ts:232]: In current mocks it is same - https://www.figma.com/file/EC2tJwlwy4RveRYhhGiHP4/Proposal-for-UI-improvement---SQL-Migration-ADS?type=whiteboard&node-id=0-1&t=CB5YVEg0XX4fijyE-0

File: extensions/sql-migration/src/wizard/skuRecommendation/skuDataCollectionToolbar.ts (hunk 2 above)
Comment (krritik) [reply extensions/sql-migration/src/wizard/skuRecommendation/skuDataCollectionToolbar.ts:20]: Done.

File: extensions/sql-migration/src/wizard/skuRecommendation/skuDataCollectionToolbar.ts [hunk 3]
```diff
@@ -14,19 +14,40 @@ import * as vscode from 'vscode';
...
+	private _startPerformanceCollectionButton!: azdata.ButtonComponent;
+	private _stopPerformanceCollectionButton!: azdata.ButtonComponent;
+	private _importPerformanceDataButton!: azdata.ButtonComponent;
+	private _recommendationParametersButton!: azdata.ButtonComponent;
+
+	private _performanceDataSource!: PerformanceDataSourceOptions;
+
 	private _disposables: vscode.Disposable[] = [];
 
+	constructor(public migrationStateModel: MigrationStateModel) {
```
Comment (stuti149) [inline extensions/sql-migration/src/wizard/skuRecommendation/skuDataCollectionToolbar.ts:33]: why migrationStateModel is public here?

File: extensions/sql-migration/src/wizard/skuRecommendation/skuDataCollectionToolbar.ts [hunk 4]
```diff
@@ -63,7 +84,47 @@ export class SkuDataCollectionToolbar implements vscode.Disposable {
...
+			if (!selectedOption || selectedOption === constants.AZURE_RECOMMENDATION_DATA_COLLECTION_DEFAULT_PATH) {
+				this._performanceDataSource = PerformanceDataSourceOptions.CollectData;
+				this.migrationStateModel._skuRecommendationPerformanceLocation = Default_PATH_FOR_START_DATA_COLLECTION;
+
+				// TODO - Start data collection at default path.
+			}
+			// 'Choose a path' option is selected.
+			else if (selectedOption === constants.AZURE_RECOMMENDATION_DATA_COLLECTION_CHOOSE_PATH) {
+				const options: vscode.OpenDialogOptions = {
+					openLabel: 'Select',
```
Comment (stuti149) [inline extensions/sql-migration/src/wizard/skuRecommendation/skuDataCollectionToolbar.ts:105]: Not sure if we need to localize this label 'Select', as this dialog is not part of ADS. But please check once from other references. I think it should be localized.

**Comment by stuti149** [comment]
I see some more text in the figma. Haven't we added that?
image

**Comment by krritik** [comment]
> I see some more text in the figma. Haven't we added that? image

So, it is limitation of ads components. We can only add one line in description. I asked Abhishek, he said it is fine if we cannot add more.

**Comment by stuti149** [comment]
> > I see some more text in the figma. Haven't we added that? image
> 
> So, it is limitation of ads components. We can only add one line in description. I asked Abhishek, he said it is fine if we cannot add more.

If it's not required then alright but we can simply change the string "Where do you want to save collected data? \n Use default path...." This will shift below text to next line.

File: extensions/sql-migration/src/wizard/skuRecommendation/skuDataCollectionToolbar.ts (hunk 3 above)
Comment (krritik) [reply extensions/sql-migration/src/wizard/skuRecommendation/skuDataCollectionToolbar.ts:33]: not needed. Changed to private.

**Comment by krritik** [comment]
> Where do you want to save collected data? **\n** Use default path....

I tried '\n'. It is not working. But for now, let's merge this PR and we can add sub-header in some next PR.

File: extensions/sql-migration/src/wizard/skuRecommendation/skuDataCollectionToolbar.ts (hunk 4 above)
Comment (krritik) [reply extensions/sql-migration/src/wizard/skuRecommendation/skuDataCollectionToolbar.ts:108]: I think we can keep it as localized. Also show something similar to be used in other extensions as well.

**Review by stuti149 — APPROVED** [review]

**Pull request merged**